File: Crazysk.java
import java.io.*;
import java.math.BigInteger;
import java.util.Scanner;
class Crazysk
{
public static int main(String args[])
{
//	BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
	Scanner sc =new Scanner(System.in);
	BigInteger one=new BigInteger("1");
	int t;
	BigInteger res,n,cards;
	t=Integer.parseInt(sc.next());
	while(t-- >0)
	{
		cards=sc.nextBigInteger();
		n=sc.nextBigInteger();
		
		res=cards;
		while((cards.add(one)).compareTo(n)==1)
		{
			res=res.add(cards.divide(n));
			cards=(cards.divide(n)).add(cards.mod(n));
			
		}
		System.out.println(res);
	}
return 0;
}
}

File: crazysk.py
import sys
t=int(input())
while(t>0):
    t=t-1;
    x,n=input().split()
    x=int(x)
    n=int(n)
    p=n
    cards=x
    res=x
    while(cards>=n):
        res =res + int(cards/n)
        cards= int(cards/n) + cards%n		
    print(res)
    
    
